pkg/server/container_stop.go

```diff
@@ -135,18 +135,16 @@ func (c *criService) stopContainer(ctx context.Context, container containerstore
 		}
 
 		sigTermCtx, sigTermCtxCancel := context.WithTimeout(ctx, timeout)
+		defer sigTermCtxCancel()
 		err = c.waitContainerStop(sigTermCtx, container)
 		if err == nil {
 			// Container stopped on first signal no need for SIGKILL
-			sigTermCtxCancel()
 			return nil
 		}
 		// If the parent context was cancelled or exceeded return immediately
 		if ctx.Err() != nil {
-			sigTermCtxCancel()
 			return ctx.Err()
 		}
-		sigTermCtxCancel()
```

Comment:
We should keep this one. The other two will run early at `return` but this one wont return until the SigKill call. This will be an early context to release `sigTermCtx` resources early.

Reply:
what happens if you call cancel twice?  

how about just calling the cancel immediately after the wait?

Reply:
> We should keep this one. The other two will run early at return but this one wont return until the SigKill call. This will be an early context to release sigTermCtx resources early.

@mikebrow I just tried Cancel twice doesn't cause issue.

@jterry75 If the code reaches here, it means that the context has timeout. :) However, I'm fine with putting cancel here to be safe. :)

Reply:
Lol good point. Too tired. :)